### ISSUE TITLE
Snaps with 0 permissions shouldn't crash the extension

### ISSUE THREAD

**Issue by FrederikBolding** [MetaMask/snaps-monorepo#899]
Snaps without any permissions don't make a lot of sense once we do #855.

An edge-case for snaps with 0 permissions should be handled. How it's handled is a matter of debate. For now let's go with allowing 0-permission snaps due to less work needed and more aligned with the permissionless ethos.

Solution: Disallow snaps with 0 permissions. Snaps without permissions cannot do anything.

### Outcome
- Fix for 0-permission snaps crashing the extension.
- E2E test for installing such a snap.

**Comment by rekmarks** [comment]
Yeah, after #855, we need to require at least one permission that makes the snap useful. It's not as simple as >0 however, because a snap with only `endowment:network-access` also can't do anything we'd be interested in supporting.

What makes a snap useful? We decide. I'm think that it can either communicate with external entitities (i.e. the RPC request permission) or do something useful to the UI (like tx insights).

**Comment by ritave** [comment]
This is important, but not needed for v1 due to allow-list
